src/System.Security.Principal.Windows/tests/SecurityIdentifierTests.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Security.Principal;
+using Xunit;
+
+public class SecurityIdentifierTests
+{
+    [Fact]
+    public void ValidateGetCurrentUser()
+    {
+        WindowsIdentity identity = WindowsIdentity.GetCurrent(false);
+        Assert.NotNull(identity.User);
+        identity?.Dispose();
+    }
+
+    [Fact]
+    public void ValidateToString()
+    {
+        string sddl = null;
+        using (WindowsIdentity me = WindowsIdentity.GetCurrent(false))
+        {
+            sddl = me.User.ToString();
+        }
+
+        Assert.NotNull(sddl);
+        Assert.NotEmpty(sddl);
+        Assert.StartsWith("S-1-5-", sddl); // sid prefix, version 1, user account type
+        Assert.NotEqual('-', sddl[sddl.Length - 1]);
+
+        string[] parts = sddl.Substring(6).Split('-');
+
+        Assert.NotNull(parts);
+        Assert.NotEmpty(parts);
+
+        Assert.All(parts, part => uint.TryParse(part, out uint _));
+    }
+
+    [Fact]
+    public void ValidateToStringUsingWhoami()
+    {
+        string librarySid = null;
+        using (WindowsIdentity me = WindowsIdentity.GetCurrent(false))
+        {
+            librarySid = me.User.ToString();
+        }
+
+        string windowsSid = null;
+        string processArguments = " /user";
+        // Call whois to get current sid
+        Process whoamiProcess = new Process();
+        whoamiProcess.StartInfo.FileName = "whoami.exe"; // whoami.exe is in system32, should already be on path
+        whoamiProcess.StartInfo.Arguments = " " + processArguments;
+        whoamiProcess.StartInfo.CreateNoWindow = true;
+        whoamiProcess.StartInfo.UseShellExecute = false;
+        whoamiProcess.StartInfo.RedirectStandardOutput = true;
+        whoamiProcess.Start();
+        string output = whoamiProcess.StandardOutput.ReadToEnd();
+        whoamiProcess.WaitForExit();
+
+        if (whoamiProcess.ExitCode == 0)
+        {
+            int startSid = output.IndexOf("S-1-5-");
+            int endSid = 0;
+            if (startSid >= 0)
+            {
+                int index = startSid + 6;
+                while (index < output.Length)
```

Comment:
This seems pretty complicated for

```C#
int length = Math.Min(output.Length - startSid, librarySid.Length);
windowsSid = output.Substring(startSid, length);

if (output.Length > startSid + length)
{
    Assert.True(char.IsWhitespace(output[startSid + length]);
}
```

Reply:
If librarySid is a prefix of the windowsSid and is different it wouldn't fail. Finding the windows sid independently of the expected value seemed safer.

Reply:
Yeah, that was the purpose of "if there's anything else the next thing better be whitespace"